File: src/main/java/com/teamwizardry/librarianlib/book/util/Link.java
package com.teamwizardry.librarianlib.book.util;

public class Link {

	
	public final String path;
	public final int page;
	
	
	public Link(String str) {
		int i = str.lastIndexOf(":");
        int page = 0;

        if (i != -1) {
            try {
                page = Integer.parseInt(str.substring(i));
            } catch (NumberFormatException e) {
                // TODO: logging
            }
        }
        String path = str.substring(0, i == -1 ? str.length() : i);
        
        this.page = page;
        this.path = path;
	}
	
	public Link(String path, int page) {
		this.path = path;
		this.page = page;
	}
	
}

File: src/main/java/com/teamwizardry/librarianlib/book/util/Link.kt
package com.teamwizardry.librarianlib.book.util

data class Link(val path:String, val tag:String)

object LinkParser {
    fun parse(link:String) : Link {
        var tag = ""
        var path = link

        val hashLoc = link.lastIndexOf("#")

        if (hashLoc != -1) {
            tag = link.substring(hashLoc)
            path = link.substring(0, hashLoc)
        }

        return Link(path, tag)
    }

}
